### PR TITLE
Sphinx doc chapter 27

### PR DESCRIPTION
**Pull request by maximedenes**

### CODE REVIEW

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 1]
```diff
@@ -1,842 +1,892 @@
...
+context of variables introduced at the beginning of the declaration,
+but the instances need not be made only of variables. Also notice that
+``A`` is *not* required to be a term having the same parameters as ``Aeq``,
+although that is often the case in practice (this departs from the
+previous implementation).
+
 In case the carrier and relations are not parametric, one can use the
-command \texttt{Add Relation} instead, whose syntax is the same except
-there is no local context.
+command ``Add Relation`` instead, whose syntax is the same except there is
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: A `cmd` is missing. There will be no index.

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 2]
```diff
@@ -1,842 +1,892 @@
...
+
+
+Adding new relations and morphisms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A parametric relation ``Aeq: forall (y1 : β1 ... ym : βm )``,
+``relation (A t1 ... tn)`` over ``(A : αi -> ... αn -> Type)`` can be
+declared with the following command:
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: That's not `coqtop` that should have been used here but `cmd`.

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 3]
```diff
@@ -1,842 +1,892 @@
...
+  Add Parametric Relation (x1 : T1) ... (xn : Tk) : (A t1 ... tn) (Aeq t′1 ... t′m )
+  [reflexivity proved by refl]
+  [symmetry proved by sym]
+  [transitivity proved by trans]
+  as id.
+
+after having required the ``Setoid`` module with the ``Require Setoid``
+command.
+
+The identifier *id* gives a unique name to the morphism and it is used
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: `texttt` should not be translated by italic

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 4]
```diff
@@ -1,842 +1,892 @@
...
- transitivity proved by (eq_set_trans (A:=A))
- as eq_set_rel.
-Add Parametric Morphism A : (@union A) with 
-signature (@eq_set A) ==> (@eq_set A) ==> (@eq_set A) as union_mor.
-Proof. exact (@union_compat A). Qed.
-\end{coq_example*}
-
-\end{cscexample}
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Should be a `cmd`

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 5]
```diff
@@ -1,842 +1,892 @@
...
+
+Some tactics (``reflexivity``, ``symmetry``, ``transitivity``) work only on
+relations that respect the expected properties. The remaining tactics
+(``replace``, ``rewrite`` and derived tactics such as ``autorewrite``) do not
+require any properties over the relation. However, they are able to
+replace terms with related ones only in contexts that are syntactic
+compositions of parametric morphism instances declared with the
+following command.
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Should be a `cmd`

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 6]
```diff
@@ -1,842 +1,892 @@
...
+
+   Require Export Setoid.
+   Require Export Relation_Definitions.
+
+.. coqtop:: in
+
+   Set Implicit Arguments.
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Why several `coqtop:: in`?

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 7]
```diff
@@ -1,842 +1,892 @@
...
+
+   Proof. apply (compat S1 S2 M). Qed.
+
+.. coqtop:: in
+
+   Lemma test: forall (S1 S2:Setoid) (m: Morphism S1 S2)
+     (x y: car S1), eq S1 x y -> eq S2 (f _ _ m x) (f _ _ m y).
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Again why all the `coqtop:: in`?

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 8]
```diff
@@ -1,842 +1,892 @@
...
-the \texttt{setoid\_rewrite} tactics directly and not \texttt{rewrite}.
+.. tacv:: Add Setoid @A @Aeq @ST as @ident
+
+where ``Aeq`` is a congruence relation without parameters, ``A`` is its carrier
+and ``ST`` is an object of type (``Setoid_Theory A Aeq``) (i.e. a record
+packing together the reflexivity, symmetry and transitivity lemmas).
+Notice that the syntax is not completely backward compatible since the
+identifier was not required.
+
+.. coqtop:: in
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Should be `cmd`

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 9]
```diff
@@ -1,842 +1,892 @@
...
+
+
+Extensions
+----------
+
+
+Rewriting under binders
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Warning: Due to compatibility issues, this feature is enabled only
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Should be `caution::` or something like that.

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 10]
```diff
@@ -1,842 +1,892 @@
...
-  \strlinea{eval}{redexpr}{apply reduction}\\
-  \strlinea{fold}{c}{fold expression}
-\end{array}\]
-
-Actually a few of these are defined in term of the others using
-a primitive fixpoint operator:
-
-\[\begin{array}{lcl}
-  \str{try}~s & = & \str{choice}~s~\str{id} \\
-  \str{any}~s & = & \str{fix}~u. \str{try}~(s~
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Was `\cite{Luttik97specificationof}`. Can't this be fixed now?

**Comment by maximedenes** [comment]
Comments addressed, thanks!

**Pull request merged**